Tests/MongoSwiftTests/MongoDatabaseTests.swift

```diff
@@ -112,26 +113,59 @@ final class MongoDatabaseTests: MongoSwiftTestCase {
         )
 
         expect(try db.createCollection("fooView", options: viewOptions)).toNot(throwError())
-        let decoder = BSONDecoder()
-        var collectionInfo = try db.listCollections().map { try decoder.decode(CollectionInfo.self, from: $0) }
+
+        var collectionInfo = try Array(db.listCollections()) as [CollectionSpecification]
         collectionInfo.sort { $0.name < $1.name }
 
         expect(collectionInfo).to(haveCount(3))
 
-        let expectedFoo = CollectionInfo(name: "foo", type: "collection", options: fooOptions)
+        let fooInfo = CollectionSpecificationInfo(readOnly: false, uuid: UUID())
+        let fooIndex = ["name": "_id_"] as Document
+        let expectedFoo = CollectionSpecification(name: "foo",
+                                                  type: .collection,
+                                                  options: fooOptions,
+                                                  info: fooInfo,
+                                                  idIndex: fooIndex)
         expect(collectionInfo[0]).to(equal(expectedFoo))
 
-        let expectedView = CollectionInfo(name: "fooView", type: "view", options: viewOptions)
+        let viewInfo = CollectionSpecificationInfo(readOnly: true, uuid: nil)
+        let expectedView = CollectionSpecification(name: "fooView",
+                                                   type: .view,
+                                                   options: viewOptions,
+                                                   info: viewInfo,
+                                                   idIndex: nil)
         expect(collectionInfo[1]).to(equal(expectedView))
 
         expect(collectionInfo[2].name).to(equal("system.views"))
     }
-}
 
-struct CollectionInfo: Decodable, Equatable {
-    let name: String
-    let type: String
-    let options: CreateCollectionOptions
+    func testListCollections() throws {
```

Comment:
I think it would be useful to add a test using command monitoring here to verify that `nameOnly: false` is sent in `listCollectionNames` and `listMongoCollections` with nil / name only filters.

To make that easier, you can use the `withCommandMonitoring` function. Check out the [change stream tests](https://github.com/mongodb/mongo-swift-driver/blob/master/Tests/MongoSwiftTests/ChangeStreamTests.swift#L548) for an example of how to use it.

Reply:
Test looks great!